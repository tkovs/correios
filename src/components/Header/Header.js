import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, StatusBar } from 'react-native'
import { Appbar, Menu } from 'react-native-paper'

import { colors } from '../../styles/theme'

const Header = ({ enableBackButton, goBack, title, subtitle }) => {
  const [menuVisible, setMenuVisible] = useState(false)

  return (
    <Appbar.Header style={styles.header}>
      {enableBackButton && <Appbar.BackAction onPress={goBack} />}
      <StatusBar backgroundColor={colors.blue} />
      <Appbar.Content title={title} subtitle={subtitle} />
      <Menu
        visible={menuVisible}
        onDismiss={() => setMenuVisible(false)}
        anchor={
          <Appbar.Action
            color="white"
            icon="dots-vertical"
            onPress={() => setMenuVisible(true)}
          />
        }
      >
        <Menu.Item onPress={() => {}} title="Editar" />
        <Menu.Item onPress={() => {}} title="Excluir" />
      </Menu>
    </Appbar.Header>
  )
}

const styles = StyleSheet.create({
  header: {
    backgroundColor: colors.blue,
  },
})

Header.defaultProps = {
  enableBackButton: false,
  goBack: null,
  subtitle: null,
}

Header.propTypes = {
  enableBackButton: PropTypes.bool,
  goBack: PropTypes.func,
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
}

export default Header
