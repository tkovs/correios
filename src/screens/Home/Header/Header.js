import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, StatusBar } from 'react-native'
import { Appbar } from 'react-native-paper'

import { colors } from '../../../styles/theme'

const Header = ({ updatePackets }) => {
  return (
    <Appbar.Header style={styles.header}>
      <StatusBar backgroundColor={colors.blue} />
      <Appbar.Content title="Correios" subtitle="Rastreio de pacotes" />
      <Appbar.Action icon="refresh" onPress={updatePackets} />
    </Appbar.Header>
  )
}

Header.propTypes = {
  updatePackets: PropTypes.func.isRequired,
}

const styles = StyleSheet.create({
  header: {
    backgroundColor: colors.blue,
  },
})

export default Header
