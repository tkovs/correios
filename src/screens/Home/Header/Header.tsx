import React, { FC } from 'react'
import { StyleSheet, StatusBar } from 'react-native'
import { Appbar } from 'react-native-paper'

import { colors } from '../../../styles/theme'

interface HeaderProps {
  updatePackets: () => void
}

type Props = HeaderProps

const Header: FC<Props> = ({ updatePackets }: Props) => (
  <Appbar.Header accessibilityStates={['disabled']} style={styles.header}>
    <StatusBar backgroundColor={colors.blue} />
    <Appbar.Content
      accessibilityStates={['disabled']}
      title="Correios"
      subtitle="Rastreio de pacotes"
    />
    <Appbar.Action
      accessibilityStates={['disabled']}
      icon="refresh"
      onPress={updatePackets}
    />
  </Appbar.Header>
)

const styles = StyleSheet.create({
  header: {
    backgroundColor: colors.blue,
  },
})

export default Header
