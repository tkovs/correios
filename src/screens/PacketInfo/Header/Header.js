import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, StatusBar } from 'react-native'
import {
  Appbar,
  Menu,
  Portal,
  Dialog,
  Paragraph,
  Button,
} from 'react-native-paper'

import { colors } from '../../../styles/theme'

const Header = ({
  addFeedback,
  archivePacket,
  goBack,
  title,
  removePacket,
}) => {
  const [menuVisible, setMenuVisible] = useState(false)
  const [dialogVisible, setDialogVisible] = useState(false)

  return (
    <Appbar.Header style={styles.header}>
      <Appbar.BackAction onPress={goBack} />
      <StatusBar backgroundColor={colors.blue} />
      <Appbar.Content title={title} subtitle="Informações do envio" />
      <Menu
        visible={menuVisible}
        onDismiss={() => setMenuVisible(false)}
        anchor={
          <Appbar.Action
            color="white"
            icon="dots-vertical"
            onPress={() => setMenuVisible(true)}
          />
        }
      >
        <Menu.Item
          onPress={() => {
            archivePacket()
            setMenuVisible(false)
          }}
          title="Arquivar"
        />
        <Menu.Item
          onPress={() => {
            setMenuVisible(false)
            setDialogVisible(true)
          }}
          title="Excluir"
        />
      </Menu>
      <Portal>
        <Dialog
          visible={dialogVisible}
          onDismiss={() => setDialogVisible(false)}
        >
          <Dialog.Title>Apagar encomenda</Dialog.Title>
          <Dialog.Content>
            <Paragraph>
              Tem certeza que deseja remover esta encomenda?
            </Paragraph>
          </Dialog.Content>
          <Dialog.Actions>
            <Button onPress={() => setDialogVisible(false)}>Não</Button>
            <Button
              onPress={() => {
                const message = 'Rastreio removido com sucesso'
                removePacket()
                setDialogVisible(false)
                addFeedback(message)
                goBack()
              }}
            >
              Sim
            </Button>
          </Dialog.Actions>
        </Dialog>
      </Portal>
    </Appbar.Header>
  )
}

const styles = StyleSheet.create({
  header: {
    backgroundColor: colors.blue,
  },
})

Header.defaultProps = {
  goBack: null,
}

Header.propTypes = {
  addFeedback: PropTypes.func.isRequired,
  archivePacket: PropTypes.func.isRequired,
  goBack: PropTypes.func,
  title: PropTypes.string.isRequired,
  removePacket: PropTypes.func.isRequired,
}

export default Header
