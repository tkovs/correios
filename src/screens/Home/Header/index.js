import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { useNavigation } from '@react-navigation/native'

import Header from './Header'
import { updatePackets } from '../../../store/actions/packets'

function HeaderContainer(props) {
  const { goBack } = useNavigation()

  return <Header {...props} goBack={goBack} />
}

HeaderContainer.propTypes = {
  updatePackets: PropTypes.func.isRequired,
}

const mapDispatchToProps = dispatch => ({
  updatePackets: () => dispatch(updatePackets()),
})

export default connect(
  null,
  mapDispatchToProps
)(HeaderContainer)
