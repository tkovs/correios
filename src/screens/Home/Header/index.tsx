import React, { FC } from 'react'
import { connect, MapDispatchToProps } from 'react-redux'

import Header from './Header'
import { updatePackets as updatePacketsAction } from '../../../store/actions/packets'
import { PacketsActionTypes } from '../../../store/actions/packets/types'

interface DispatchProps {
  updatePackets: () => PacketsActionTypes
}

type Props = DispatchProps

const HeaderContainer: FC<Props> = ({ updatePackets }: Props) => {
  return <Header updatePackets={updatePackets} />
}

const mapDispatchToProps: MapDispatchToProps<DispatchProps, {}> = (
  dispatch
) => ({
  updatePackets: () => dispatch(updatePacketsAction()),
})

export default connect(null, mapDispatchToProps)(HeaderContainer)
