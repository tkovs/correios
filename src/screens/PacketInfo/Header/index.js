import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { useNavigation } from '@react-navigation/native'
import { removePacket, archivePacket } from '../../../store/actions/packets'
import { addFeedback } from '../../../store/actions/feedback'

import Header from './Header'

function HeaderContainer(props) {
  const { archivePacket: archive, removePacket: remove, code, ...rest } = props
  const { goBack } = useNavigation()

  return (
    <Header
      {...rest}
      goBack={goBack}
      removePacket={() => remove(code)}
      archivePacket={() => archive(code)}
    />
  )
}

HeaderContainer.propTypes = {
  removePacket: PropTypes.func.isRequired,
  archivePacket: PropTypes.func.isRequired,
  code: PropTypes.string.isRequired,
}

const mapDispatchToProps = dispatch => ({
  removePacket: code => dispatch(removePacket(code)),
  archivePacket: code => dispatch(archivePacket(code)),
  addFeedback: message => dispatch(addFeedback(message)),
})

export default connect(
  null,
  mapDispatchToProps
)(HeaderContainer)
