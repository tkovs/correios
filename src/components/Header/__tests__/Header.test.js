import 'react-native'
import React from 'react'
import { render } from '@testing-library/react-native'

import Header from '../Header'

it('renders correctly', () => {
  const { baseElement } = render(
    <Header title="Correios" subtitle="Rastreio de pacotes" />
  )
  expect(baseElement).toMatchSnapshot()
})
