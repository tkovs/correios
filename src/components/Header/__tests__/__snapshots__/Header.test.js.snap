// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header just with title renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView
      style={
        Array [
          Object {
            "backgroundColor": "#01417F",
            "elevation": 4,
            "zIndex": 0,
          },
          Object {
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#01417F",
            "elevation": 0,
            "flexDirection": "row",
            "height": 56,
            "marginTop": 0,
            "paddingHorizontal": 4,
          }
        }
      >
        <TouchableWithoutFeedback>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 12,
                },
                Array [
                  Object {
                    "marginLeft": 8,
                  },
                  Object {
                    "alignItems": "center",
                  },
                  undefined,
                ],
              ]
            }
          >
            <Text
              accessibilityRole="header"
              accessibilityTraits="header"
              accessible={true}
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "fontSize": 17,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Correios
            </Text>
          </View>
        </TouchableWithoutFeedback>
        <View
          collapsable={false}
        >
          <TouchableHighlight
            accessibilityComponentType="button"
            accessibilityRole="button"
            accessibilityStates={Array []}
            accessibilityTraits="button"
            activeOpacity={0.85}
            centered={true}
            delayPressOut={100}
            disabled={false}
            hitSlop={
              Object {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "margin": 6,
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 18,
                    "height": 36,
                    "width": 36,
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
            underlayColor="rgba(255, 255, 255, 0.16)"
          >
            <View>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    Object {
                      "height": 24,
                      "width": 24,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "rotate": "0deg",
                        },
                      ],
                    }
                  }
                >
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 24,
                        },
                        Array [
                          Object {
                            "transform": Array [
                              Object {
                                "scaleX": 1,
                              },
                            ],
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                        ],
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    ï‡™
                  </Text>
                </View>
              </View>
            </View>
          </TouchableHighlight>
        </View>
      </View>
    </SafeAreaView>
  </View>
</View>
`;

exports[`Header with title and subtitle renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView
      style={
        Array [
          Object {
            "backgroundColor": "#01417F",
            "elevation": 4,
            "zIndex": 0,
          },
          Object {
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#01417F",
            "elevation": 0,
            "flexDirection": "row",
            "height": 56,
            "marginTop": 0,
            "paddingHorizontal": 4,
          }
        }
      >
        <TouchableWithoutFeedback>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 12,
                },
                Array [
                  Object {
                    "marginLeft": 8,
                  },
                  Object {
                    "alignItems": "center",
                  },
                  undefined,
                ],
              ]
            }
          >
            <Text
              accessibilityRole="header"
              accessibilityTraits="header"
              accessible={true}
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "fontSize": 17,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Correios
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "fontSize": 11,
                    },
                    Object {
                      "color": "rgba(255, 255, 255, 0.7)",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Rastreio de pacotes
            </Text>
          </View>
        </TouchableWithoutFeedback>
        <View
          collapsable={false}
        >
          <TouchableHighlight
            accessibilityComponentType="button"
            accessibilityRole="button"
            accessibilityStates={Array []}
            accessibilityTraits="button"
            activeOpacity={0.85}
            centered={true}
            delayPressOut={100}
            disabled={false}
            hitSlop={
              Object {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "margin": 6,
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 18,
                    "height": 36,
                    "width": 36,
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
            underlayColor="rgba(255, 255, 255, 0.16)"
          >
            <View>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    Object {
                      "height": 24,
                      "width": 24,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "rotate": "0deg",
                        },
                      ],
                    }
                  }
                >
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 24,
                        },
                        Array [
                          Object {
                            "transform": Array [
                              Object {
                                "scaleX": 1,
                              },
                            ],
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                        ],
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    ï‡™
                  </Text>
                </View>
              </View>
            </View>
          </TouchableHighlight>
        </View>
      </View>
    </SafeAreaView>
  </View>
</View>
`;
